org.eclipse.m2e.launching/src/org/eclipse/m2e/internal/launch/MavenConsoleLineTracker.java

```diff
@@ -69,59 +75,63 @@
 
   private static final String RUNNING_MARKER = "Running ";
 
-  private static final String TEST_TEMPLATE = "(?:  )test.+\\(([\\w\\.]+)\\)"; //$NON-NLS-1$
+  private static final Pattern TEST_CLASS_PATTERN = Pattern.compile("(?:  )test.+\\(([\\w\\.]+)\\)"); //TODO: what does this match?
```

Comment:
I wonder what kind of lines this Pattern is intended to match? Lines that contain any fully qualified class name that starts with `test` and is prefixed with two spaces?
If yes the Pattern seems to be wrong, because I suspect the first dot should have been escaped, so the following capturing group likely only captured one symbol.
Furthermore if the fully-qualified class name starts with `test.` one will not find it without that first name element.

For me it looks like it would only work if a Maven-plugin prints `  test.`+test-class-name for a executed test-class.
Does anybody know such plugin?
If not, I suggest to remove that Pattern entirely.

Reply:
I believe this pattern is meant to capute testMethod (starting by `test`, `.+` being the other names of the method, and the other part capturing paramters. This is probably used in order to like to method when some test failures are reported.
IMO, we should keep it, however we should improve it (later) so that it doesn't rely on the assumption that test methods start by `test`.